datetime/src/main/java/com/fasterxml/jackson/datatype/jsr310/deser/InstantDeserializer.java

```diff
@@ -285,6 +288,20 @@ private ZoneId getZone(DeserializationContext context)
         return (_valueClass == Instant.class) ? null : context.getTimeZone().toZoneId();
     }
 
+    private String replaceZeroOffsetAsZIfNecessary(String text)
+    {
+        if (replaceZeroOffsetAsZ && endsWithZeroOffset(text)) {
```

Comment:
This seems to evaluate regexp twice, first to match, then to replace: would it be possible to make that just one operation? Regexp evaluation has non-trivial cost, even considering that textual date handling itself is rather expensive operation.